src/sage/schemes/elliptic_curves/ell_finite_field.py

```diff
@@ -379,21 +453,195 @@ def random_element(self):
+                if (p - 1) % order != 0:
+                    raise ValueError(f"the curve does not have a point of order {order}")
+                cofactor = (p - 1) // order
+            else:
+                raise NotImplementedError("not implemented for ordinary curves or supersingular curves with cyclic group of points")
+
+            while True:
+                P = self.random_element()
+                P = cofactor * P
+
+                P.set_order(multiple=order)
+                P_order = generic.order_from_multiple(P, order)
+                if P_order == order:
+                    P.set_order(P_order)
+                    return P
+
+        elif alg == "divPol":
+            if order == 1:
+                return self.zero()
+
+            points = []
+
+            for (ell, e) in order.factor():
+                divpol = self.division_polynomial(ell)
+                roots = divpol.numerator().roots(multiplicities=False)
+                shuffled_indices = list(range(len(roots)))
+                shuffle(shuffled_indices)
+
+                if not roots:
+                    raise ValueError(f"the curve does not have a point of order {order}")
+
+                for index in shuffled_indices:
+                    if ell == 2:
+                        # y-coodinate is bound to be 0
+                        P = self(roots[index], 0)
+
+                    else:
+                        # Random sampling of a x-coordinate and a choice for the y coordinate
+                        Ps = self.lift_x(roots[index], all=True)
+                        if Ps:
+                            i = ZZ.random_element(2)
+                            P = Ps[i]
+                        else:
+                            continue
+
+                    for _ in range(e - 1):
+                        divpoints = P.division_points(ell)
+                        if not divpoints:
+                            break
+
+                        i = ZZ.random_element(len(divpoints))
+                        P = divpoints[i]
+                    else:
+                        points.append(P)
+                        break
+
+                else:
+                    # Fail when all torsion points have rational x-,
+                    # but irrational y-coordinate
+                    raise ValueError(f"the curve does not have a point of order {order}")
+
+            return sum(points)
+
+        elif alg == "abelianGroup":
+            generators = self.gens()
+
+            if len(generators) == 2:
+                (P,Q) = generators
+            elif len(generators) == 1:
+                P = generators[0]
+                Q = self.zero()
+            else:
+                raise ValueError(f'the curve does not have any generators defined over the base field')
+
+            # Multiply P, Q by an appropriate cofactor s.t. both have the correct order
+            if P.order() % order != 0 or Q.order() % order != 0:
```

Comment:
If only one generator exists, this will incorrectly raise an error.
```suggestion
            if P.order() % order != 0 or (Q and Q.order() % order != 0):
```